### PR TITLE
Development

### PR DESCRIPTION
**Pull request by FuhrmannMaciej**
Set UI for debug console, refactored code and is now able to add and use commands.

### CODE REVIEW

File: Runtime/Scripts/DeveloperConsole.cs [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+            if (Input.GetKeyDown(KeyCode.BackQuote))
+            {
+                consoleCanvas.gameObject.SetActive(!consoleCanvas.gameObject.activeInHierarchy);
+            }
+
+            if (consoleCanvas.gameObject.activeInHierarchy)
+            {
+                if (Input.GetKeyDown(KeyCode.Return))
+                {
+                    if (inputText.text != "")
```
Comment (0xLeif) [inline Runtime/Scripts/DeveloperConsole.cs]: Could you just have one if statement? 

`if (consoleCanvas.gameObject.activeInHierarchy && Input.GetKeyDown(KeyCode.Return) && inputText.text != "") { ... }`

File: Runtime/Scripts/DeveloperConsole.cs [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+                return;
+            }
+
+            if (!Commands.ContainsKey(input[0]))
+            {
+                Debug.LogWarning("Command not recognized.");
+            }
+            else
+            {
+                Commands[input[0]].RunCommand();
```
Comment (0xLeif) [inline Runtime/Scripts/DeveloperConsole.cs]: Why only use the first input?

File: Runtime/Scripts/DeveloperConsole.cs [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+    public abstract class ConsoleCommand
+    {
+        public abstract string Name { get; protected set; }
+        public abstract string Command { get; protected set; }
+        public abstract string Description { get; protected set; }
+        public abstract string Help { get; protected set; }
+
+        public void AddCommandToConsole()
+        {
+            DeveloperConsole.AddCommandsToConsole(Command, this);
```
Comment (0xLeif) [inline Runtime/Scripts/DeveloperConsole.cs]: 🤔 Weird that a ConsoleCommand knows about a Singleton...

File: Runtime/Scripts/DeveloperConsole.cs [hunk 4]
```diff
@@ -0,0 +1,145 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace Oneleif.debugconsole
+{
+    public abstract class ConsoleCommand
```
Comment (0xLeif) [inline Runtime/Scripts/DeveloperConsole.cs]: This could be an Interface

File: Runtime/Scripts/Commands/CommandQuit.cs [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+            }
+            else
+            {
+                Application.Quit();
+            }
+        }
+
+        public static CommandQuit CreateCommand()
+        {
+            return new CommandQuit();
```
Comment (0xLeif) [inline Runtime/Scripts/Commands/CommandQuit.cs]: 🤔 This seems a bit unnecessary ...

File: Runtime/Scripts/DeveloperConsole.cs [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+            string message = string.Empty;
+            if (type.Equals(LogType.Warning))
+            {
+                message = "<color=yellow>[" + type.ToString() + "]" + logMessage + "</color> ";
+            }
+            else
+            {
+                message = "<color=white>[" + type.ToString() + "]" + logMessage + "</color> ";
+            }
+            AddMessageToConsole(message);
```
Comment (0xLeif) [inline Runtime/Scripts/DeveloperConsole.cs]:
```
string color = type.Equals(LogType.Warning) ? "yellow" : "white";
AddMessageToConsole("<color=" + color + ">[" + type.ToString() + "]" + logMessage + "</color> ");
```

**Review by 0xLeif — APPROVED** [review]

**Review by ambid17 — CHANGES_REQUESTED** [review]
this code does not need to be merged to master

**Comment by FuhrmannMaciej** [comment]
closing PR for now

**Pull request closed without merging**